### PR TITLE
Update MongoDB perl driver

### PR DESCRIPTION
**Pull request by hangy**
Updates the MongoDB Perl driver to v2.2, and replaces calls to removed subs with alternatives.

### CODE REVIEW

**Comment by hangy** [comment]
The most important change is that the previously deprecated method `MongoDB::Cursor::count()` has been deprecated now. Therefore, some code had to be changed to either count manually, or after eagerly retrieving `my @products = $cursor->all;`.

**Comment by stephanegigandet** [comment]
The save() method has been removed? There's a bunch of script that still use it:

ns3135968:~/hangy/openfoodfacts-server/scripts$ grep collection *.pl | grep save
credit_product_to_creator.pl:                   get_products_collection()->save($product_ref);
fix_code_stored_as_number.pl:                   $products_collection->save($product_ref);
fix_leading_zeros.pl:                                           get_products_collection()->save($product_ref);
fix_product2.pl:get_products_collection()->save($product_ref);
fix_product.pl:         get_products_collection()->save($product_ref);
remove_deleted_products_from_db.pl:             # get_products_collection()->save($product_ref);
remove_products_without_code.pl:                # get_products_collection()->save($product_ref);
scanbot.pl:                     get_products_collection()->save($product_ref);
test_additifs.pl:                       #get_products_collection()->save($product_ref);
update_all_products.pl:                         get_products_collection()->save($product_ref);

**Comment by VaiTon** [comment]
Also the entire driver is gonna be in end of life on 2020. From https://metacpan.org/changes/distribution/MongoDB:
> Version v2.2.0 is the final feature release of the MongoDB Perl
      driver.  The driver is now in a 12-month "sunset" period and will
      receive security patches and critical bug fixes only.  The Perl
      driver will be end-of-life and unsupported on August 13, 2020.

**Comment by stephanegigandet** [comment]
@hangy : the changes I made in master that caused the conflicts can be removed (the "my $cursor = get_products_collection()->find($query_ref, { projection => { code => 1, id => 1, _id => 1 }});;"), I was trying to debug something. Although it might be useful to restrict the fields returned.

File: scripts/obsolete/update_all_products_all_fields.pl [hunk 1]
```diff
@@ -50,10 +50,7 @@
 # Get a list of all products
 
 
-my $cursor = $products_collection->query({})->fields({ code => 1 });;
-my $count = $cursor->count();
-	
-	print STDERR "$count products to update\n";
```
Comment (stephanegigandet) [inline scripts/obsolete/update_all_products_all_fields.pl]: I'll put back the count in this script, I look at it all the time. :)

**Comment by stephanegigandet** [comment]
For reference, to update the Perl MongoDB driver through cpan: https://github.com/mongodb/mongo-perl-driver/blob/master/lib/MongoDB/Upgrading.pod#installing-v2-over-v1

**Pull request merged**